coursier.bzl

```diff
@@ -562,6 +568,31 @@ def _coursier_fetch_impl(repository_ctx):
         False,  # not executable
     )
 
+    # Generate a compatibility layer of external repositories for all jar artifacts.
+    if repository_ctx.attr.generate_compat_repositories:
+        compat_repositories_bzl = ["def compat_repositories():"]
+        for versionless_target_label in jar_imports:
+            repository_ctx.file(
+                versionless_target_label + "/WORKSPACE",
+                "",
+                executable = False,
+            )
+            repository_ctx.file(
+                versionless_target_label + "/jar/BUILD",
+                "alias(name = \"jar\", actual = \"@maven//:%s\", visibility = [\"//visibility:public\"])" % versionless_target_label,
+                executable = False,
+            )
+            compat_repositories_bzl.append("    native.local_repository(")
+            compat_repositories_bzl.append("        name = \"%s\"," % versionless_target_label)
+            compat_repositories_bzl.append("        path = \"%s\"," % repository_ctx.path(versionless_target_label))
```

Comment:
Instead of using local_repository in this hacky way, please add a new repository rule implementation of which is just generating the right BUILD file